go/filemanager/pfsserver/handler.go

```diff
@@ -0,0 +1,328 @@
+package pfsserver
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+
+	pfsmod "github.com/PaddlePaddle/cloud/go/filemanager/pfsmodules"
+	sjson "github.com/bitly/go-simplejson"
+	log "github.com/golang/glog"
+)
+
+type response struct {
+	Err     string      `json:"err"`
+	Results interface{} `json:"results"`
+}
+
+func cmdHandler(w http.ResponseWriter, req string, cmd pfsmod.Command) {
+	resp := response{}
+
+	if err := cmd.CloudCheck(); err != nil {
+		resp.Err = err.Error()
+		writeJSONResponse(w, req, http.StatusOK, &resp)
+		return
+	}
+
+	result, err := cmd.Run()
+	if err != nil {
+		resp.Err = err.Error()
+		resp.Results = result
+		writeJSONResponse(w, req, http.StatusOK, &resp)
+		return
+	}
+
+	resp.Results = result
+	writeJSONResponse(w, req, http.StatusOK, &resp)
+}
+
+func lsHandler(w http.ResponseWriter, r *http.Request) {
+	cmd, err := pfsmod.NewLsCmdFromURLParam(r.URL.RawQuery)
+
+	resp := response{}
+	if err != nil {
+		resp.Err = err.Error()
+		writeJSONResponse(w, r.URL.RawQuery, http.StatusOK, &resp)
+		return
+	}
+
+	cmdHandler(w, r.URL.RawQuery, cmd)
+}
+
+func statHandler(w http.ResponseWriter, r *http.Request) {
+	log.V(1).Info("begin stathandler")
+	cmd, err := pfsmod.NewStatCmdFromURLParam(r.URL.RawQuery)
+
+	resp := response{}
+	if err != nil {
+		resp.Err = err.Error()
+		writeJSONResponse(w, r.URL.RawQuery, http.StatusOK, &resp)
+		return
+	}
+
+	cmdHandler(w, r.URL.RawQuery, cmd)
+}
+
+func writeJSONResponse(w http.ResponseWriter,
+	req string,
+	httpStatus int,
+	resp *response) {
```

Comment:
Maybe pass by value is better `resp response`. Please see https://github.com/golang/go/wiki/CodeReviewComments#pass-values